### DISCUSSION TITLE
AAR submit APIs (/data/receive.php) returns 404

### DISCUSSION THREAD

**Discussion by VyMajoris** [Zealot111/OCAP#11] [Q&A]
**OCAP Version:** `3.2.0-stable`
**Server OS:** `Windows`

**Description:** When the addon calls the extention to upload the recording, the server returns 404.  

![image](https://user-images.githubusercontent.com/4342296/121436660-617f4300-c957-11eb-843d-0ca43ce65959.png)
The server is runnig. I can access my AAR webpage.

**Steps to reproduce:** Call ocap_fnc_exportData

**ocap-web options.json**:
```json
{
	"title": "OCAP",
	"description": "Operation Capture And Playback",
	"author": "Frontend: MisterGoodson (aka Goodson [3CB]), Backend: Dell",
	"language": "en",
	"version": "3.2.0",
	"listen": "127.0.0.1:5000",
	"secret": "secret",
	"logger": true,
	"classes-game": [
		{"key": "", "name": "All"},
		{"key": "tvt", "name": "TvT"}
	]
}
```
**OcapReplaySaver2.cfg.json**:
```json
{
    "addFileUrl": "http://127.0.0.1:5000/data/receive.php?option=addFile",
    "dbInsertUrl": "http://127.0.0.1:5000/data/receive.php?option=dbInsert",
    "httpRequestTimeout": 120,
    "newMode": 0,
    "newServerGameType": "",
    "newUrl": "https://127.0.0.1:5000/api/v1/operations/add",
    "newUrlRequestSecret": "secret",
    "traceLog": 1
}
```
**userconfig/ocap.hpp**:
```json
ocap_nameServer = 'Test';
ocap_minPlayerCount = 1;
ocap_frameCaptureDelay = 1;
ocap_excludeFromRecord = ["ACE_friesAnchorBar"];
ocap_saveMissionEnded = true;
ocap_minMissionTime = 1;
````

**RPT File**
[ocap-ext.20210609_222355.log](https://github.com/Zealot111/OCAP/files/6627223/ocap-ext.20210609_222355.log)
[ocap-main.20210609_222355.log](https://github.com/Zealot111/OCAP/files/6627224/ocap-main.20210609_222355.log)

**Side note:** 
Please provide a quick documentaion on what these values mean and where they are used:
**classes-game, newMode, newServerGameType, ocap_nameServer**

**Comment by Zealot111** [comment]
@VyMajoris  why `newmode` is set to 0 ? Is it intensional? 
`"newMode" : 1` means that dll send mission using `"newUrl": "https://127.0.0.1:5000/api/v1/operations/add"`
(intended for use with new go web service, which is included in installation)
(and not using `addFileUrl` and `dbInsertUrl` which vere used for php scripts in previous versions )
You should set `"newMode": 1,` unless you want something very specific

**Comment by Zealot111** [comment]
Ocap allow several type of missions to be filtered by user, for example you may have 'tvt1' arma3 instance, 'tvt2' instance and so on. User will be able to filter only tvt2 missions in ocap for example.
So you may configure ocap to distinguish missions from these servers.
Try default value and change only if you want something specific
default value - `"newServerGameType": "tvt",` `ocap_nameServer = 'TvT';`

**Comment by Zealot111** [comment]
```
	"classes-game": [
		{"key": "", "name": "All"},
		{"key": "tvt", "name": "TvT"}
	]
```
describes these instance names, only `tvt` in default configuration

**Comment by VyMajoris** [comment]
Thanks. To be honest I'm not sure why `newmode` was set to 0. I probably copied from an older config somewhere in our server.
I changed it to 1 and it worked. I just had to also change the `newUrl` "https" to "http".

**Comment by Zealot111** [comment]
@VyMajoris please consider starring repository if you find our project useful ⭐
